### PR TITLE
Migrate to typescript

### PR DESCRIPTION
**Pull request by sirahd**
- add types to the original javascript file
- upgrades dependency for `d3` and adds `browserify, tsify`

### CODE REVIEW

File: src/options.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+export interface Option {
+  showAllFields?: boolean,
+  fields?: Field[],
+  delay?: number,
+  offset?: {
+    x: number,
+    y: number
+  },
+  onAppear?(event: Event, item: any): void,
```
Comment (kanitw) [inline src/options.ts]: With TS2.2 you can use `object` instead of `any` :)

Comment (kanitw) [inline src/options.ts]: (For the case that you know it's really an arbitrary object.)

File: src/vg-tooltip.ts [hunk 2]
```diff
@@ -0,0 +1,850 @@
...
+
+      // supplement the fieldOption with fieldDef and config
+      var supplementedFieldOption = supplementFieldOption(fieldOption, fieldDef, vlSpec);
+
+      supplementedFields.push(supplementedFieldOption);
+    });
+  }
+  // if showAllFields is false, only supplement existing fields in options.fields
+  else {
+    if (options.fields && options.fields.length > 0) {
```
Comment (kanitw) [inline src/vg-tooltip.ts]: No need to check `options.fields.length > 0`   (if your length is zero, forEach won't run)

File: src/options.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  onMove?(event: Event, item: any): void,
+  onDisappear?(event: Event, item: any): void,
+  colorTheme?: 'light' | 'dark'
+}
+
+export interface Field extends FieldDef {
```
Comment (kanitw) [inline src/options.ts]: This should be named FieldOption?

**Comment by kanitw** [comment]
LGTM. is it working now?

**Comment by sirahd** [comment]
@kanitw Done!

**Pull request merged**